### ISSUE TITLE
Changing an item of a loaded set, also overwrites the set in localStorage ("by reference" issue)

### ISSUE THREAD

**Issue by klingo** [watchyoursixx/HunterSim#55]
### Steps to reproduce:
1. Load a custom set
2. Change one item
3. Load the same custom set again

### Expected behavior:
"Old" item from the saved set is shown

### Actual behavior:
The "new" changed item is still there

### Root cause analysis:
When loading a (custom) set, in '**loadSet()**' the set is read from "SavedSets" and a **reference** to its object is put in the '**gear**' variable.
```js
let activeset = SavedSets[val];
gear = activeset.data;
```

When an item is changed and '**selectItem(itemid)**' is called, it overwrites the value(s) in '**gear**'.
```js
case 'shoulder':
    console.log("you selected "+ itemid);
    gear[activeslot] = { id: parseInt(itemid), gems: [], enchant: ench };
break;
```
Now since '**gear**' is basically just a **reference** to the '**SavedSets[val].data**' object, it is also changed there.

------------------
### Simplified example and solution proposal:
Simple example to reproduce/understand this issue:
```js
const myArray = [
    {
        'name': 'First Entry'
    },
    {
        'name': 'Second Entry'
    }
];

let firstObject = myArray[0]; // firstObject is just a reference to the first entry in the Array

console.log(myArray[0].name); // "First Entry"

firstObject.name = 'Changed Entry';

console.log(myArray[0].name); // "Changed Entry"
```

In order to fix it, you need to make a shallow/deep copy of the set when loading it. In case of above example, the following would result in a deep-copy:
```js
const myArray = [
    {
        'name': 'First Entry'
    },
    {
        'name': 'Second Entry'
    }
];

let firstObject = JSON.parse(JSON.stringify(myArray[0])); // now an actual (deep) copy of the object is created

console.log(myArray[0].name); // "First Entry"

firstObject.name = 'Changed Entry';

console.log(myArray[0].name); // "First Entry"
```

**Comment by watchyoursixx** [comment]
Thank you for the detailed info!

I actually did this in "saveSet()" and "addSet()" functions, but somehow missed it in the "LoadSet()". Was wondering why it was still doing it after having added those...

**Comment by watchyoursixx** [comment]
Fixed in #56
